### PR TITLE
Blog app - Add forms

### PR DESCRIPTION
**Pull request by bobb-Rob**
## I implemented the following:
- Create a method called current_user in your ApplicationController to make current user data available to all controllers.
- It will return the first user from the database.
- Create forms to perform the following functions:
      - Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
      - Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
      - Allow Users to add likes to Posts.

### CODE REVIEW

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @bobb-Rob, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request
- [ ] Your project has an error that is preventing it from running
image

You have submitted a project, that <EXPLAIN WHY PROJECT IS INVALID>

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class CommentsController < ApplicationController
+  def create
+    @comment = current_user.comments.new(post_params)
+    @comment.post_id = params[:post_id]
+    @comment.save
+    redirect_to user_post_path(@comment.author_id, @comment.post_id)
```
Comment (mikethreels) [inline app/controllers/comments_controller.rb]: - [ ] in this method you should check if the comment was saved like below:
```
    if @comment.save
      redirect_to post_path(@comment.post)
    else
      @article = Post.find(params[:post_id])
      redirect_to post_path(@post)
    end
```

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Robertson, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job following the instructions of the previous CR

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (bobb-Rob) [reply app/controllers/comments_controller.rb]: Thanks for the review, but I don't understand the else block code  @article = Post.find(params[:post_id])
      redirect_to post_path(@post),.....What does this do and where is the @post in post_path(@post) coming from?

Comment (bobb-Rob) [reply app/controllers/comments_controller.rb]: @mikethreels And what about the @article.

**Review by FrederickMih — APPROVED** [review]
Hi @bobb-Rob, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :green_circle: 

- Commit messages are descriptive and capitalized :+1: 
- PR is professional :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (mikethreels) [reply app/controllers/comments_controller.rb]: Hi @bobb-Rob, sorry for the delay, I see you solved the issue in the meantime as your project is approved. Congratulations!!

**Pull request merged**